aquacomputer_d5next.c

```diff
@@ -760,14 +818,27 @@ static int aqc_legacy_read(struct aqc_data *priv)
 
 	/* Temperature sensor readings */
 	for (i = 0; i < priv->num_temp_sensors; i++) {
-		sensor_value = get_unaligned_be16(priv->buffer + priv->temp_sensor_start_offset +
+		sensor_value = get_unaligned_le16(priv->buffer + priv->temp_sensor_start_offset +
 						  i * AQC_TEMP_SENSOR_SIZE);
-		if (sensor_value == AQC_SENSOR_NA)
-			priv->temp_input[i] = -ENODATA;
-		else
-			priv->temp_input[i] = sensor_value * 10;
+		priv->temp_input[i] = sensor_value * 10;
 	}
 
+	sensor_value = (s16)get_unaligned_le16(priv->buffer + priv->fan_sensor_offsets[0]);
+	priv->speed_input[0] = aquastream_convert_pump_rpm(sensor_value);
+
+	sensor_value = (s16)get_unaligned_le16(priv->buffer + priv->fan_sensor_offsets[1]);
+	priv->speed_input[1] = aquastream_convert_fan_rpm(sensor_value);
+
+	/* Calculation derived from linear regression */
+	priv->current_input[0] = get_unaligned_le16(priv->buffer + AQUASTREAMXT_PUMP_CURR_OFFSET)
+				 * 176 / 100 - 52;
```

Comment:
I think it's better to use DIV_ROUND_CLOSEST() here and elsewhere whenever something is being divided.